backend/conversationParser.js

```
@@ -0,0 +1,305 @@
+
+  // Remove markdown artifacts that are just formatting
+  // Keep actual code blocks and formatting that adds meaning
+
+  // Remove excessive newlines (more than 2)
+  cleaned = cleaned.replace(/\n{3,}/g, '\n\n');
+
+  // Trim whitespace
+  cleaned = cleaned.trim();
+
+  // Remove any null bytes or other control characters
+  cleaned = cleaned.replace(/[\x00-\x08\x0B-\x0C\x0E-\x1F\x7F]/g, '');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)